pkg/detection/compiled_policy.go

```diff
@@ -62,10 +63,16 @@ func newCompiledPolicy(policy *storage.Policy) (CompiledPolicy, error) {
 				"one runtime constraint from either process, or kubernetes event category, or network baseline.", policy.GetName())
 		}
 		// set predicates
-		compiled.predicates = append(compiled.predicates, &deploymentPredicate{scopes: scopes, exclusions: exclusions})
+		// Only add deployment predicate for deployment-based runtime events, not for node events
+		if policy.GetEventSource() != storage.EventSource_NODE_EVENT {
```

Comment:
This might be more readable and easier to understand as a switch case with the order of cases reversed as compared to the ifs.
case storage.EventSource_NODE_EVENT:
case storage.EventSource_AUDIT_LOG_EVENT:
default